File: ClientDashboard_API/DTOs/CompleteTrainerAnalyticsDto.cs
using ClientDashboard_API.Records;

namespace ClientDashboard_API.DTOs
{
    public class CompleteTrainerAnalyticsDto
    {
        // CLIENT METRICS
        public int BaseClients { get; set; }

        public int AcquiredClients { get; set; }

        public double AcquisitionPercentage { get; set; }

        public int ChurnedClients { get; set; }

        public double ChurnPercentage { get; set; }

        public int NetGrowth { get; set; }

        public double NetGrowthPercentage { get; set; }

        public int SessionsPerClient { get; set; }

        public int MonthlyClientSessions { get; set; }

        // REVENUE PATTERNS

        public decimal SessionsPrice { get; set; }

        public int MonthlyWorkingDays { get; set; }

        public double RevenuePerWorkingDay { get; set; }

        public double RevenuePerWorkingWeek { get; set; }

        public double RevenuePerWorkingMonth { get; set; }

        // ACTIVITY PATTERNS

        public required List<WeeklyMultiplier> BusiestDays { get; set; }

        public required List<WeeklyMultiplier> LightDays { get; set; }
    }
}

File: ClientDashboard_GUI/src/app/models/dtos/complete-trainer-analytics-dto.ts
import { WeeklyActivityPattern } from './weekly-activity-pattern';

export interface CompleteTrainerAnalyticsDto {
  baseClients: number;
  acquiredClients: number;
  acquisitionPercentage: number;
  churnedClients: number;
  churnPercentage: number;
  netGrowth: number;
  netGrowthPercentage: number;
  averageSessionsPerClient: number;
  totalClientSessions: number;
  sessionsPrice: number;
  monthlyWorkingDays: number;
  totalRevenue: number;
  revenuePerWorkingDay: number;
  revenuePerWorkingWeek: number;
  totalWorktimeMinutes: number;
  averageDailyWorktime: number;
  averageWeeklyWorktime: number;
  allWeekdays: WeeklyActivityPattern[];
  busiestDays: WeeklyActivityPattern[];
  lightDays: WeeklyActivityPattern[];
}